pkg/endpoint/bpf.go

```diff
@@ -613,6 +624,18 @@ func (e *Endpoint) regenerateBPF(owner Owner, currentDir, nextDir string, regenC
 			return 0, compilationExecuted, fmt.Errorf("unable to regenerate policy because PolicyMap synchronization failed: %s", err)
 		}
 
+		// Synchronously update the BPF ConfigMap for this endpoint.
+		// This is unlikely to fail, but will have the same
+		// inconsistency issues as above if there is a failure. Long
+		// term the solution to this is to templatize this map in the
+		// ELF file, but there's no solution to this just yet.
+		if err = e.bpfConfigMap.Update(e.desiredBPFConfig); err != nil {
```

Comment:
This is weird because we are doing this update and everyone is happy. What happens if line 642 fails? Are we reverting this update?

Reply:
I pushed a function onto the `RevertStack` in the latest upload.